core/src/main/java/org/elasticsearch/http/netty/NettyHttpChannel.java

```diff
@@ -134,6 +137,7 @@ public void sendResponse(RestResponse response) {
             if (!addedReleaseListener && content instanceof Releasable) {
                 ((Releasable) content).close();
             }
+            transport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
```

Comment:
I assume that you want `NettyHttpChannel` to implement `AutoCloseable`. I think that would just make sense if we would use `NettyHttpChannel` in a try-with-resources block. I can however implement `Closeable` and add a precondition check there that we do not close it twice.